pages/partials/question.ejs

```diff
@@ -199,6 +199,36 @@
         <% }); %>
       </div>
 
+      <script>
+        $(function() {
+          $('.js-submission-body.render-pending').on('show.bs.collapse', function(e) {
+            const submissionId = $(this).data('submission-id');
+            // <% if (question_context === 'instructor') { %>
+            const submissionContentUrl = `<%= urlPrefix %>/question/<%= question.id %>/preview/variant/<%= variant.id %>/submission/${submissionId}`;
+            // <% } else if (question_context === 'manual_grading') { %>
+            const submissionContentUrl = `<%= urlPrefix %>/assessment/<%= assessment.id %>/manual_grading/instance_question/<%= instance_question.id %>/variant/<%= variant.id %>/submission/${submissionId}`;
+            // <% } else { %>
+            const submissionContentUrl = `<%= urlPrefix %>/instance_question/<%= instance_question.id %>/variant/<%= variant.id %>/submission/${submissionId}`;
+            // <% } %>
+            const wasModalOpen = ($(`#submissionInfoModal-${submissionId}`).data('bs.modal') || {})._isShown;
+            $(`#submissionInfoModal-${submissionId}`).modal('hide');
+            $.get(submissionContentUrl, 'json')
+              .done((data) => {
+                $(`#submission-${submissionId}`).replaceWith($(data.submissionPanel).fadeIn('slow'));
```

Comment:
Should this `replaceWith` call have a lifting function inside instead of a value, like the other calls to `replaceWith` in `pages/partials/externalGradingLiveUpdate.ejs` do?

Reply:
I don't see a reason to do so here. A function, in my view, makes sense if you're replacing more than one element and the replacement depends on the element being replaced, or if the computation would be extensive and may not happen if the element doesn't exist. Neither is the case here.

Reply:
Is there any difference with the calls in `externalGradingLiveUpdate.ejs`? It looks like the ones there are only using the function to lambda lift (make the evaluation lazier). But I could be wrong about that.

Reply:
I don't think jQuery does lazy evaluation here, but I could be wrong. My understanding is that the function is called immediately for each element being replaced.